modules/ray_tracing/src/raytracing/ClaimRays.C

```diff
@@ -209,3 +219,94 @@ ClaimRays::buildPointNeighbors()
     }
   }
 }
+
+void
+ClaimRays::verifyClaiming()
+{
+  // NOTE for all of the following: we use char here in place of bool.
+  // This is because bool is not instantiated as a StandardType in
+  // TIMPI due to the fun of std::vector<bool>
+
+  // Map from Ray ID -> whether or not it was generated (false) or
+  // claimed/possibly also generated (true)
+  std::map<RayID, char> local_map;
+  auto add_to_local_map = [this, &local_map](const std::vector<std::shared_ptr<Ray>> & rays,
+                                             const bool claimed_rays) {
+    for (const auto & ray : rays)
+    {
+      const auto id = getID(ray);
+
+      // Try to insert into the map
+      auto emplace_pair = local_map.emplace(id, claimed_rays);
+
+      // Already existed in the map
+      if (!emplace_pair.second)
+      {
+        // If it already exits but has not been claimed yet, set it to being claimed
+        if (claimed_rays && !emplace_pair.first->second)
+          emplace_pair.first->second = true;
+        // Otherwise, it is being doubly generated/claimed
+        else
+          mooseError("ClaimRays for ",
+                     _study.errorPrefix(),
+                     ": Ray with ID ",
+                     id,
+                     " was ",
+                     claimed_rays ? "claimed" : "generated",
+                     " multiple times on pid ",
+                     _pid,
+                     "\n\n",
+                     ray->getInfo());
+      }
+    }
+  };
+
+  // Build the local_map
+  add_to_local_map(_rays, false);
+  add_to_local_map(_local_rays, true);
+
+  // Build the structure to send the local generation/claiming information to rank 0
+  std::map<processor_id_type, std::vector<std::pair<RayID, char>>> send_info;
+  send_info.emplace(std::piecewise_construct,
+                    std::forward_as_tuple(0),
+                    std::forward_as_tuple(local_map.begin(), local_map.end()));
+
+  // The mapping (filled on rank 0) from Ray ID -> (processor id, claiming status)
+  std::map<RayID, std::vector<std::pair<processor_id_type, char>>> global_map;
+
+  // Functor for reciving the generation/claiming information
+  auto receive_functor =
+      [&global_map](processor_id_type pid,
+                    const std::vector<std::pair<RayID, char>> & id_claimed_pairs) {
+        for (const auto & id_claimed_pair : id_claimed_pairs)
+          global_map[id_claimed_pair.first].emplace_back(pid, id_claimed_pair.second);
+      };
+
+  // Send claiming information to rank 0
+  Parallel::push_parallel_vector_data(comm(), send_info, receive_functor);
+
+  // Rank 0 will make sure everything looks good
+  if (_pid == 0)
+    for (const auto & id_pairs_pair : global_map)
+    {
+      const RayID id = id_pairs_pair.first;
+      const std::vector<std::pair<processor_id_type, char>> & pid_claimed_pairs =
+          id_pairs_pair.second;
+
+      std::vector<processor_id_type> claimed_pids;
```

Comment:
the pids arent claimed, they claim. Having trouble finding a better name though

Reply:
🤷🏼‍♂️ close enough